compiler/src/iree/compiler/Dialect/HAL/Conversion/HALToVM/ConvertFenceOps.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Dialect/HAL/IR/HALOps.h"
+#include "iree/compiler/Dialect/VM/Conversion/ImportUtils.h"
+#include "iree/compiler/Dialect/VM/IR/VMOps.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+namespace {
+
+struct FenceCreateOpConversion
+    : public OpConversionPattern<IREE::HAL::FenceCreateOp> {
+  FenceCreateOpConversion(MLIRContext *context, SymbolTable &importSymbols,
+                          TypeConverter &typeConverter, StringRef importName)
+      : OpConversionPattern(context) {
+    importOp = importSymbols.lookup<IREE::VM::ImportOp>(importName);
+    assert(importOp);
+  }
+  LogicalResult matchAndRewrite(
+      IREE::HAL::FenceCreateOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    auto importType = importOp.getFunctionType();
+
+    SmallVector<Value, 8> callOperands;
```

Comment:
For later: I really felt myself reaching for some kind of CallBuilder for doing this kind of thing...

Reply:
Agreed!